model.py

```diff
@@ -1211,16 +1218,114 @@ class Hyperlink(Base):
     __tablename__ = 'hyperlinks'
 
     id = Column(Integer, primary_key=True)
-    library_id = Column(Integer, ForeignKey('libraries.id'), index=True)
     rel = Column(Unicode, index=True, nullable=False)
-    href = Column(Unicode, nullable=False)
+    library_id = Column(Integer, ForeignKey('libraries.id'), index=True)
+    resource_id = Column(Integer, ForeignKey('resources.id'), index=True)
 
     # A Library can have multiple links with the same rel, but we only
     # need to keep track of one.
     __table_args__ = (
         UniqueConstraint('library_id', 'rel'),
     )
 
+    @hybrid_property
+    def href(self):
+        if not self.resource:
+            return None
+        return self.resource.href
+
+    @href.setter
+    def href(self, url):
+        _db = Session.object_session(self)
+        resource, is_new = get_one_or_create(_db, Resource, href=url)
+        self.resource = resource
+
+
+class Resource(Base):
+    """A URI, potentially linked to multiple libraries, or to a single
+    library through multiple relationships.
+
+    e.g. a library consortium may use a single email address as the
+    patron help address and the integration contact address for all of
+    its libraries. That address only needs to be validated once.
+    """
+    __tablename__ = 'resources'
+
+    id = Column(Integer, primary_key=True)
+    href = Column(Unicode, nullable=False, index=True, unique=True)
+    hyperlinks = relationship("Hyperlink", backref="resource")
+
+    # Every Resource may have at most one Validation. There's no
+    # need to validate it separately for every relationship.
+    validation_id = Column(Integer, ForeignKey('validations.id'),
+                           index=True)
+    validation = relationship(
+        "Validation", backref=backref("resource", uselist=False)
+    )
```

Comment:
I think our convention so far has been to put the relationship on the model class that doesn't have the id column.

It may not make a difference, but that way you can still see that there's a relationship when you only look at the Validation class.